go/libkb/deprovision.go

```diff
@@ -0,0 +1,65 @@
+package libkb
+
+import (
+	"fmt"
+)
+
+// XXX: THIS DELETES SECRET KEYS. Deleting the wrong secret keys can make you
+// lose all your data forever. We only run this in the DeprovisionEngine and if
+// we detect that our device was revoked in LogoutAndDeprovisionIfRevoked.
+func ClearSecretsOnDeprovision(m MetaContext, username NormalizedUsername) error {
+	// 1. Delete all the user's secret keys!!!
+	// 2. Delete the user's ephemeralKeys
+	// 3. Delete the user from the config file.
+	// 4. Db nuke.
```

Comment:
this list only have 4 items and the one above has 5?

Reply:
Do you also clear out the active device?  Maybe  you should hold `g.switchUsersMu` while you do this. See `libkb/context.go`

Reply:
callers take care of logout (if we detect a revoke and logout or call `attemptLoggedInRevoke` in the deprovision engine)

i'll add active device clearing, is there anything else that would be good to clear here ?

Reply:
Cool, thanks.

That's about all I can think of.  Seems like you need a combo of `SwitchUserNukeConfig` and `SwitchUserLoggedOut`. Maybe make `SwitchUserDeprovision` and just copy-pasta.

Reply:
it does, just checked.

Reply:
ok cool ill just add the `switchUsersMu` while we nuke the config then